src/components/store/SizeSelector/index.tsx

```diff
@@ -0,0 +1,56 @@
+import styles from '@/components/store/SizeSelector/style.module.scss';
+import { PublicMerchItemOption } from '@/lib/types/apiResponses';
+import { useEffect, useId, useState } from 'react';
+
+interface SizeSelectorProps {
+  currSize: string | undefined;
+  options: PublicMerchItemOption[];
+  // Justification for disabling rules: This seems to be a false positive.
+  // https://stackoverflow.com/q/63767199/
+  // eslint-disable-next-line no-unused-vars
+  setSize: (value: string) => void;
+}
+
+const SizeSelector = ({ currSize, options, setSize }: SizeSelectorProps) => {
+  const [mounted, setMounted] = useState(false);
+  const myID = `${useId()}`;
```

Comment:
I think `useId` already returns a string so you don't need to cast it into a string again